Basic/Gen/PP.pm

```diff
@@ -3499,9 +3505,10 @@ $PDL::PP::deftbl =
 			my $dimobjs  = $_[2];
 
 			$result->[1]{"SIZE"} = sub {
-			  croak "can't get SIZE of undefined dimension (RedoDims=$redodims)."
-			    unless defined $dimobjs->{$redodims};
-			  return $dimobjs->{$redodims}->get_size();
+			    eval 'use PDL::IO::Dumper';
+			    croak "FOO can't get SIZE of undefined dimension (RedoDims=$redodims).\nredodims is $redodims\ndimobjs is ".sdump($dimobjs)."\n"
+			       unless defined $dimobjs->{$_[0]};  # This is the closure's $_[0], not the rule definition's $_[0]
```

Comment:
Wouldn't this be better as something like:

```
			    # This is the closure's $_[0], not the rule definition's $_[0]
			    unless (defined $dimobjs->{$_[0]}) {
			      require PDL::IO::Dumper;
			      croak "FOO can't get SIZE of undefined dimension (RedoDims=$redodims).\nredodims is $redodims\ndimobjs is ".PDL::IO::Dumper::sdump($dimobjs)."\n";
			    }
```

This would avoid loading `PDL::IO::Dumper` unless there were an actual error, which would also make things a bit quicker.

Also, is it possible to add a test of this `SIZE` behaviour?

Reply:
```suggestion
			    # This is the closure's $_[0], not the rule definition's $_[0]
			    if (!defined $dimobjs->{$_[0]}) {
			      require PDL::IO::Dumper;
			      croak "FOO can't get SIZE of undefined dimension (RedoDims=$redodims).\nredodims is $redodims\ndimobjs is ".PDL::IO::Dumper::sdump($dimobjs)."\n";
			    }
```

Reply:
I think I looked at this and couldn't even figure out whether or where the SIZE behavior was actually used in PDL.  Closing and merging was the right thing to do here.